### PR TITLE
Added c-tor with explicit log levels param

### PR DESCRIPTION
**Pull request by starkmsu**

### CODE REVIEW

File: src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs [hunk 1]
```diff
@@ -22,16 +25,42 @@ public class LykkeLogToAzureSlackNotificationsManager : ProducerConsumer<LogEnti
             _slackNotificationsSender = slackNotificationsSender;
             _lastResortLog = lastResortLog ?? new LogToConsole();
             _component = componentName;
+            _logLevels = new List<string>
```
Comment (MikhailGerasimov) [inline src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: Use an array

File: src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs [hunk 2]
```diff
@@ -22,16 +25,42 @@ public class LykkeLogToAzureSlackNotificationsManager : ProducerConsumer<LogEnti
...
         public LykkeLogToAzureSlackNotificationsManager(
             ISlackNotificationsSender slackNotificationsSender,
+            IEnumerable<string> logLevels,
             ILog lastResortLog = null)
             : base(lastResortLog)
         {
             _slackNotificationsSender = slackNotificationsSender;
             _lastResortLog = lastResortLog ?? new LogToConsole();
             _component = AppEnvironment.Name;
+            _logLevels = logLevels ?? new List<string>(0);
```
Comment (MikhailGerasimov) [inline src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: Use Array.Empty<>

File: src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs [hunk 3]
```diff
@@ -22,16 +25,42 @@ public class LykkeLogToAzureSlackNotificationsManager : ProducerConsumer<LogEnti
...
+
+        public LykkeLogToAzureSlackNotificationsManager(
+            ISlackNotificationsSender slackNotificationsSender,
+            ILog lastResortLog = null)
+            : base(lastResortLog)
+        {
+            _slackNotificationsSender = slackNotificationsSender;
+            _lastResortLog = lastResortLog ?? new LogToConsole();
+            _component = AppEnvironment.Name;
+            _logLevels = new List<string>
```
Comment (MikhailGerasimov) [inline src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: Extract duplicated code of _logLevelsInitialization

File: src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs [hunk 4]
```diff
@@ -43,10 +72,7 @@ public void SendNotification(LogEntity entry)
         {
             try
             {
-                if (entry.Level != LykkeLogToAzureStorage.ErrorType
-                    && entry.Level != LykkeLogToAzureStorage.FatalErrorType
-                    && entry.Level != LykkeLogToAzureStorage.WarningType
-                    && entry.Level != LykkeLogToAzureStorage.MonitorType)
+                if (_logLevels.All(l => l != entry.Level))
```
Comment (MikhailGerasimov) [inline src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: Using index (as enum is an integer type) could be faster but maybe harder to understand

File: src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs [hunk 5]
```diff
@@ -12,6 +14,7 @@ public class LykkeLogToAzureSlackNotificationsManager : ProducerConsumer<LogEnti
         private readonly ISlackNotificationsSender _slackNotificationsSender;
         private readonly ILog _lastResortLog;
         private readonly string _component;
+        private readonly IEnumerable<string> _logLevels;
```
Comment (polonzer) [inline src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: I would use HashSet for better performance.

Comment (MikhailGerasimov) [reply src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: Not always better for small integer arrays. GetHashCode() can cost more then iterating through the array

File: src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs [hunk 6]
```diff
@@ -42,25 +36,19 @@ public class LykkeLogToAzureSlackNotificationsManager : ProducerConsumer<LogEnti
...
-                LykkeLogToAzureStorage.WarningType,
-                LykkeLogToAzureStorage.MonitorType,
-            };
+            _logLevels = DefaultLogLevelsInit();
         }
 
         public LykkeLogToAzureSlackNotificationsManager(
             ISlackNotificationsSender slackNotificationsSender,
-            IEnumerable<string> logLevels,
+            HashSet<string> logLevels,
```
Comment (MikhailGerasimov) [inline src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: ISet interface is better

Comment (starkmsu) [reply src/Lykke.Logs/LykkeLogToAzureSlackNotificationsManager.cs]: In ISet there is no clear Contains method. No clear advantage against Hashset,

**Review by MikhailGerasimov — APPROVED** [review]

**Pull request merged**